### ISSUE TITLE
Caikit API Docs

### ISSUE THREAD

**Issue by joerunde** [caikit/caikit#69]
## Description

Caikit python references on readthedocs.org which describe Caikit API.

## Discussion

We've used sphinx in the past with some success. We had been heavily relying on the `autodoc` features to walk the import tree and document everything, but the resulting docs were a bit scattered and sometimes missing.

It might be worth explicitly annotating which apis should be documented, and laying that out in a docs tree for sphinx to generate instead.

## Acceptance Criteria

- [x] Add mechanism to ganerate API docs (#125)
- [ ] caikit refs on [readthedocs.org](https://docs.readthedocs.io) (#154)

**Comment by hickeyma** [comment]
What about [pydoc](https://docs.python.org/3/library/pydoc.html)?

**Comment by hickeyma** [comment]
Re-opening as still need to host on readthedoc.org

**Comment by afrittoli** [comment]
I was able to generate the python docs locally with `tox -e docs` which is really nice.
I guess this issue is about publishing those to readthedocs on PR merge.

I checked the [readthedocs](https://docs.readthedocs.io/en/stable/tutorial/) and they need a GitHub account with enough privileges in the github org. It might be nice to create a functional GitHub account for that, unless anyone of the maintainers would like to use their own personal account for this.
